sources/packages/backend/apps/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -226,13 +224,35 @@ export class ApplicationStudentsController extends BaseController {
    * @param applicationId application id.
    * @returns application warnings.
    */
-  @Get(":applicationId/get-application-warnings")
+  @Get(":applicationId/warnings")
+  @ApiNotFoundResponse({
+    description:
+      "Applications does not exists or the student does not have access to it.",
+  })
   async getApplicationWarnings(
     @Param("applicationId", ParseIntPipe) applicationId: number,
-  ): Promise<ECertFailedValidation[]> {
-    const eCertFailedValidationsPromise =
-      this.eCertPreValidationService.executePreValidations(applicationId, true);
-    return eCertFailedValidationsPromise;
+    @UserToken() userToken: StudentUserToken,
+  ): Promise<ApplicationWarningsAPIOutDTO> {
+    const applicationExist = await this.applicationService.doesApplicationExist(
```

Comment:
API endpoint was missing student authorization and also returning an object not converted to a DTO.